app/controllers/chat_channels_controller.rb

```diff
@@ -177,9 +177,11 @@ def render_unopened_ids_response
   end
 
   def render_joining_request_json_response
-    requested_memberships = current_user.chat_channel_memberships.includes(:chat_channel).
-      where(chat_channels: { discoverable: true }, role: "mod").pluck(:chat_channel_id).map { |membership_id| ChatChannel.find_by(id: membership_id).requested_memberships }
-    render json: { joining_requests: requested_memberships.flatten }
+    requested_memberships_id = current_user.chat_channel_memberships.includes(:chat_channel).
+      where(chat_channels: { discoverable: true }, role: "mod").pluck(:chat_channel_id).map { |membership_id| ChatChannel.find_by(id: membership_id).requested_memberships }.flatten.map(&:id)
+    @chat_channels_memberships = ChatChannelMembership.includes(%i[user chat_channel]).where(id: requested_memberships_id)
+
+    render "index.json"
```

Comment:
this is quite hard to read, can we split it in multiple lines? For example:

```ruby
    requested_memberships_id = current_user.chat_channel_memberships.
      includes(:chat_channel).
      where(chat_channels: { discoverable: true }, role: "mod").
      pluck(:chat_channel_id).map do |membership_id|
        ChatChannel.find_by(id: membership_id).requested_memberships
      end.flatten.map(&:id)
    @chat_channels_memberships = ChatChannelMembership.includes(%i[user chat_channel]).where(id: requested_memberships_id)

    render "index.json"
```

Looking at the query I'm also thinking if there is no way to simplify it with:

```ruby
chat_channel_ids = current_user.chat_channel_memberships.includes(:chat_channel).where(chat_channels: { discoverable: true }, role: "mod").pluck(:chat_channel_id)
@chat_channel_memberships = ChatChannelMembership.includes(%i[user chat_channel]).where(status: "joining_request", chat_channel_id: chat_channel_ids)
```

am I missing something?

Reply:
So this code will be obsolete after next PR. So we didn't optimised it.

Reply:
ah ok!